cirq-core/cirq/circuits/circuit_operation_test.py

```diff
@@ -928,4 +946,64 @@ def test_simulate_no_repetition_ids_inner(sim):
     assert result.records['1:a'].shape == (1, 2, 1)
 
 
+@pytest.mark.parametrize('sim', [cirq.Simulator(), cirq.DensityMatrixSimulator()])
+def test_repeat_until(sim):
+    q = cirq.LineQubit(0)
+    key = cirq.MeasurementKey('m')
+    c = cirq.Circuit(
+        cirq.CircuitOperation(
+            cirq.FrozenCircuit(
+                cirq.X(q) ** 0.2,
```

Comment:
Just use regular `X` and make sure that it exits after two loops (this should allow Clifford simulation too).